src/main/java/com/vmware/vipclient/i18n/base/instances/TranslationMessage.java

```diff
@@ -96,67 +104,141 @@ public String getMessage(final Locale locale, final String component, final Stri
+                dto.setLocale(locale.toLanguageTag());
+                String mt = s.postString(dto);
                 if (VIPCfg.getInstance().isMachineTranslation() && !"".equalsIgnoreCase(mt)) {
                     translation = mt;
                 }
             }
         }
 
-        if (!VIPCfg.getInstance().isMachineTranslation() && VIPCfg.getInstance().isPseudo() &&
-                null != translation && translation.equals(source)) {
+        if (!VIPCfg.getInstance().isMachineTranslation() && VIPCfg.getInstance().isPseudo()
+                && null != translation && translation.equals(source)) {
             // if source isn't collected by server, add PSEUDOCHAR2
             translation = ConstantsKeys.PSEUDOCHAR2 + translation + ConstantsKeys.PSEUDOCHAR2;
         }
 
-        if (args != null && args.length > 0) {
-            if ((null != translation && translation.equals(source)) || VIPCfg.getInstance().isPseudo()) {
-                translation = FormatUtils.formatMsg(translation,
-                        LocaleUtility.getDefaultLocale(), args);
-            } else {
-                translation = FormatUtils.formatMsg(translation, locale, args);
+        if (args != null) {
+            if (args instanceof Object[] && ((Object[]) args).length > 0) {
+                if ((null != translation && translation.equals(source)) || VIPCfg.getInstance().isPseudo()) {
+                    translation = FormatUtils.formatWithArrayArgs(translation, LocaleUtility.getDefaultLocale(), (Object[]) args);
+                } else {
+                    translation = FormatUtils.formatWithArrayArgs(translation, locale, (Object[]) args);
+                }
+            } else if (args instanceof Map<?, ?> && ((Map<?, ?>) args).size() > 0) {
+                if ((null != translation && translation.equals(source)) || VIPCfg.getInstance().isPseudo()) {
+                    translation = FormatUtils.formatWithNamedArgs(translation,
+                            LocaleUtility.getDefaultLocale(), (Map<String, Object>) args);
+                } else {
+                    translation = FormatUtils.formatWithNamedArgs(translation, locale, (Map) args);
+                }
             }
         }
+
+        return translation;
+    }
+
+    private String getStringWithoutSource(MessagesDTO dto, final Object args) {
+        Locale locale = Locale.forLanguageTag(dto.getLocale());
+        String translation = new StringService().getString(dto);
+        if (StringUtil.isEmpty(translation)) {
+            return "";
+        }
+
+        if (args != null) {
+            if (args instanceof Object[] && ((Object[]) args).length > 0) {
+                translation = FormatUtils.format(translation, locale, (Object[]) args);
+            } else if (args instanceof Map<?, ?> && ((Map) args).size() > 0) {
+                translation = FormatUtils.formatMsg(translation, locale, (Map<String, Object>) args);
+            }
```

Comment:
This code block appear multiple times in this class, so I think it can be wrapped as a method and put into FormatUtil class, then the code lines in this class can be reduced and code looks more clear.